sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -196,17 +201,58 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
-    return this.repo.update(
-      {
-        msfaaNumber: msfaaNumber,
-        cancelledDate: IsNull(),
-        newIssuingProvince: IsNull(),
-      },
-      {
-        cancelledDate: getISODateOnlyString(cancelledDate),
-        newIssuingProvince,
-      },
-    );
+    return this.dataSource.transaction(async (transactionEntityManager) => {
+      // Keeping both MSFAA Cancellation update and saving of MSFAA Notification message as a part of the same transaction.
+      // Update the MSFAA record that needs to be cancelled.
+      const updateResult = await this.repo.update(
+        {
+          msfaaNumber: msfaaNumber,
+          cancelledDate: IsNull(),
+          newIssuingProvince: IsNull(),
+        },
+        {
+          cancelledDate: getISODateOnlyString(cancelledDate),
+          newIssuingProvince,
+        },
+      );
+      // In case no MSFAA record was cancelled.
+      if (!updateResult.affected) {
```

Comment:
we are throwing the error on the consumers function, so, do we need it here? 
![image](https://github.com/bcgov/SIMS/assets/77353155/3bd96701-2a39-461e-9300-9ea761b19ce7)

Reply:
I have removed it from the consumer to keep it with `msfaa-number.service.ts` generating the `updateResult`.